fleet/api/http/metrics.go

```diff
@@ -16,81 +19,329 @@ type metricsMiddleware struct {
 	counter metrics.Counter
 	latency metrics.Histogram
 	svc     fleet.Service
+	auth    mainflux.AuthServiceClient
 }
 
-func (m metricsMiddleware) ViewOwnerByChannelIDInternal(ctx context.Context, channelID string) (fleet.Agent, error) {
+func (m metricsMiddleware) ViewOwnerByChannelIDInternal(ctx context.Context, channelID string) (agent fleet.Agent, _ error) {
+	defer func(begin time.Time) {
+		labels := []string{
+			"method", "viewOwnerByChannelIDInternal",
+			"owner_id", agent.MFOwnerID,
```

Comment:
Probably these fields are empty. You should change the order of the m.svc.ViewOwnerByChannelIDInternal, calling it before you span the metrics. You're using agent, but it's the returned object.

Reply:
So we should verify this - because this is run `defer` I think there might be a value, since it should be running after the function is all finished

Reply:
You're right... just need to confirm. I think I face empty values on redis producer, but it was not running `defer`.

Reply:
I had checked this earlier, here is an example, it's not the same function but the same approach and we're able get the value this way. See the value of index 3
![image](https://user-images.githubusercontent.com/86990690/143284468-0549e6c7-bfc6-4515-b564-8e93f447be0a.png)

Reply:
You're are completly right, I also tested here with same results. My bad... I did remove the other comments.